pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -220,8 +227,9 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 				// Remote side closed connection.
 				return receiveError
 			}
-			// This should be only set for the first request. Guard with ID check regardless.
-			if discReq.Node != nil && discReq.Node.Id != "" {
+			// This should be only set for the first request. Must be called even if the request has
+			// a missing node id (for example native gRPC)
```

Comment:
Agreed on user agent, we shouldn't use it - but we can use node metadata to change the behavior.

However - this is not about gRPC. Any client ( like an attacker ) sending an empty node ID would crash Istiod. 

For gRPC - we can set proper node ID ( see the test ).

Reply:
The comment on this line was more general - no problem with the node ID change

Reply:
gRPC should always set the node in the initial request.  If you see some evidence that we're not doing so, please file a bug.

Reply:
It doesn't matter if gRPC sends or not - Istio should not crash if anyone connecting to it doesn't set a node ( or send any intentionally invalid config ). It's a security DOS risk. 

And gRPC does send a node - but the Id field may be empty if the bootstrap doesn't include one. Which was crashing istiod instead of rejecting or defaulting to sane behavior ( since we no longer depend on ID so much - we use labels ).